### PR TITLE
feat/loginandsignup

### LINKED ISSUE #11: As a user, I want to see a login page when I open the website
### **BACKEND**
The login page should have a username/password form and a forgot password button. 

To do:

- Create the relevant functions in app.py that will render the templates created in the front-end ticket. 
- Create the `class LoginForm(FlaskForm)` that will accept an email `StringField` and a password `PasswordField`

Parent Story: https://github.com/gianfrancodumoulinbertucci/SOEN341/issues/10#issue-801664122

### LINKED ISSUE #12: As a user, I want to see a sign up page where I can create an account
sign up page should provide email, username password form in order for a user to create an account

Parent Story: https://github.com/gianfrancodumoulinbertucci/SOEN341/issues/10#issue-801664122

### PR DESCRIPTION
**Pull request by mohamad-elchanati**
This feature will be responsible for the backend and frontend of login and sign up pages

### CODE REVIEW

**Review by david-roper — APPROVED** [review]
This looks good, 5 stars

File: app/app.py [hunk 1]
```diff
@@ -84,18 +93,125 @@ def image_post():
...
+  #------------------------------------------------------  
+class Register(FlaskForm):
+    username = StringField('username', validators=[InputRequired()])
+    email = EmailField('email', validators=[InputRequired(), Email()])
+    firstname = StringField('firstname', validators=[InputRequired()])
+    lastname = StringField('lastname', validators=[InputRequired()])
+    password = PasswordField('password', validators=[InputRequired(), Length(max=18,min=6)])
+    #submit = SubmitField('Submit')
+
+usernameinfo = ''
```
Comment (gianfrancodumoulinbertucci) [inline app/app.py:105]:
```suggestion
usernameinfo = ''
firstnameinfo = ''
lastnameinfo = ''
emailinfo = ''
```
I think this was left in by accident, delete if not needed

File: app/app.py [hunk 2]
```diff
@@ -84,18 +93,125 @@ def image_post():
...
+firstnameinfo = ''
+lastnameinfo = ''
+emailinfo = ''
```
Comment (gfdb) [inline app/app.py]:
```suggestion
usernameinfo = ''
firstnameinfo = ''
lastnameinfo = ''
emailinfo = ''
```
These aren't used anywhere, delete if not needed, if needed disregard this

File: app/app.py [hunk 3]
```diff
@@ -84,18 +93,125 @@ def image_post():
...
+class Login(FlaskForm):
+    username = StringField('username', validators=[InputRequired()])
+    password = PasswordField('password', validators=[InputRequired()])
+    #submit = SubmitField('Submit')
```
Comment (gfdb) [inline app/app.py]: move to forms.py

File: app/app.py [hunk 4]
```diff
@@ -84,18 +93,125 @@ def image_post():
...
 
-    
+  #------------------------------------------------------  
+class Register(FlaskForm):
+    username = StringField('username', validators=[InputRequired()])
+    email = EmailField('email', validators=[InputRequired(), Email()])
+    firstname = StringField('firstname', validators=[InputRequired()])
+    lastname = StringField('lastname', validators=[InputRequired()])
+    password = PasswordField('password', validators=[InputRequired(), Length(max=18,min=6)])
+    #submit = SubmitField('Submit')
```
Comment (gfdb) [inline app/app.py]: move to forms.py

File: app/app.py [hunk 5]
```diff
@@ -84,18 +93,125 @@ def image_post():
...
 
+            username = form.username.data
+            session['username'] = form.username.data
+            flash(username + " is logged in",category='username is logged in')
+            return redirect(url_for('timeline'))
 
-    
-    
+        else:
+            flash("Your USERNAME/PASSWORD might be incorrect or you are not a warrior !" ,category='loginerror')
```
Comment (gfdb) [inline app/app.py]: change message to "Username or password is incorrect"

File: app/static/accounts.json [hunk 6]
```diff
@@ -0,0 +1,44 @@
+[
+    {
+        "email": "concordia@shit.ca",
```
Comment (gianfrancodumoulinbertucci) [inline app/static/accounts.json:3]: ??????????????

File: app/app.py [hunk 7]
```diff
@@ -84,18 +93,125 @@ def image_post():
...
+    return render_template("login.html", form=form, posts=posts)
+
+
+'''xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'''
```
Comment (gfdb) [inline app/app.py]: delete

File: app/app.py [hunk 8]
```diff
@@ -84,18 +93,125 @@ def image_post():
...
+    with open('app/static/accounts.json', 'r') as accounts_file:
+                accounts = load(accounts_file)
+
+    for account in accounts:
+        if username == account['username'] and password == account['password']:
+                
+            return True
+
+
+
```
Comment (gfdb) [inline app/app.py]: remove white space

**Review by gfdb — CHANGES_REQUESTED** [review]
some things need to be fixed

**Review by gfdb — APPROVED** [review]
Looks good, well done

**Comment by mohamad-elchanati** [comment]
login and signup are now functional

**Comment by mohamad-elchanati** [comment]
login and signup are now functional

**Comment by mohamad-elchanati** [comment]
login and signup are now functional

**Pull request merged**